pkg/kubectl/cmd/logs.go

```diff
@@ -199,7 +225,40 @@ func (o LogsOptions) Validate() error {
 
 // RunLogs retrieves a pod log
 func (o LogsOptions) RunLogs() (int64, error) {
-	req, err := o.LogsForObject(o.Object, o.Options)
+	switch t := o.Object.(type) {
```

Comment:
Why this code isn't part of the `LogsForObject` defined in `pkg/kubectl/cmd/util/factory.go`?